compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -168,6 +168,8 @@ pub struct Elaborator<'context> {
 
     /// Temporary flag to enable the experimental arithmetic generics feature
     enable_arithmetic_generics: bool,
+
+    pub(crate) interpreter_call_stack: Vec<Location>,
```

Comment:
```suggestion
    pub(crate) interpreter_call_stack: im::Vector<Location>,
```
We should use `im::Vector` here for call stacks like we do in SSA